tests/test_importantsource.py

```diff
@@ -0,0 +1,606 @@
+#!/usr/bin/env python3
+# Copyright (C) 2018 Endless Mobile, Inc.
+#
+# Authors:
+#       Joaquim Rocha <jrocha@endlessm.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import datetime as dt
+import logging
+import os
+import unittest
+import tempfile
+import hashlib
+import base64
+from xml.dom import minidom
+
+import aiohttp
+
+from src.lib.utils import init_logging
+from src.lib.checksums import MultiDigest
+from src.lib.errors import CheckerFetchError
+from src.manifest import ManifestChecker
+from src.lib.externaldata import ExternalGitRepo
+from src.checkers.gitchecker import GitChecker
+
+
+class DummyChecker(GitChecker):
+    def get_json_schema(self, external_data):
+        return None
+
+    @classmethod
+    def should_check(cls, external_data):
+        return True
+
+    async def check(self, external_data):
+        logging.debug(
+            "Phony checker checking external data %s and all is always good",
+            external_data.filename,
+        )
+
+
+class UpdateEverythingChecker(DummyChecker):
```

Comment:
Ideally, the tests for important source should be merged into `test_checker.py` and reusing the common code instead of just copying it over.

Reply:
Hm, I had previously split off the tests into a new file due to https://github.com/flathub/flatpak-external-data-checker/pull/276#pullrequestreview-893415396. 

Would you still prefer re-combining the files (and I assume also the classes)?

The tests combined are about a 1000 lines long. Also, the `DummyChecker` in `test_importantsource.py` subclasses `GitChecker` since the tests I wrote happened to use git sources. 

The tests could be switched to `extra-data` sources. Or, `test_checker.py`'s `DummyChecker` could be switched to subclass `GitChecker` (but I'm unsure if it's 100% correct for the existing `extra-data` tests in `test_checker.py`).

Reply:
>Would you still prefer re-combining the files (and I assume also the classes)?

My initial suggestion was to move the important source tests into separate class, not entire separate file. I still think it would be better, but no strong opinion on that.

Reply:
> I still think it would be better, but no strong opinion on that.

I've combined the tests files, and split off `_test_update` so it can be reused by both sets of tests. There is now only a few lines of copy paste (the git checker subclassing). I needed to slightly adjust the exisiting _test_update calls, but they look correct.